runtime/include/chpl-tasks-callbacks.h

```diff
@@ -0,0 +1,161 @@
+//   type chpl_task_cb_info_t, describing the task and/or event.  This
+//   type is also defined below.
+//
+//
+// RETURN VALUE
+//
+//   The returned value is 0 if no errors occurred and some other value
+//   (with errno set) if any errors did occur.
+//
+//
+// ERRORS
+//
+//   The following errors can occur with chpl_task_install_callback():
+//
+//     ENOMEM:  No room to install another callback function for this
+//              event.  At present there is a static limit of 10 installed
+//              callback functions for each event.
+//     ERANGE:  The specified 'kind' is too large.
+//
+//   The following errors can occur with chpl_task_uninstall_callback():
+//
+//     ENOENT:  The given pointer was not found in the list of installed
+//              callback functions for the given event.
+//     ERANGE:  The specified 'kind' is too large.
+//
+//
+// NOTES
+//
+//   Task callbacks are local to the top-level locale.  In multi-locale
+//   programs, callback functions must be installed separately on each
+//   locale where they are wanted.
+//
+//   The task callback system is not thread safe.  Calling the install
+//   and/or uninstall functions simultaneously from more than one thread
+//   on a single locale can corrupt internal data structures and lead to
+//   chaos.  The installed callback functions themselves may be called
+//   by more than one thread on the same locale simultaneously; if they
+//   need concurrency control they must provide for it themselves.
+//
+//   The callback function should not depend upon the allocation
+//   status of the pointed-to info or any member within it, past the
+//   point when it returns.  For example, it must not store a copy of
+//   the info pointer of an event_kind==chpl_task_cb_event_kind_begin
+//   call, for use when the corresponding chpl_task_cb_event_kind_end
+//   call occurs.  It should also not store a copy of the filename
+//   pointer from an info_kind==full call, to be used in a subsequent
+//   info_kind==id_only call.  If the info as a whole or the filename
+//   are wanted later, the callback function must allocate memory to
+//   hold a copy of the pointed-to data and duplicate it itself.
+//
+
+typedef enum {
+  chpl_task_cb_event_kind_create,
+  chpl_task_cb_event_kind_begin,
+  chpl_task_cb_event_kind_end,
+  chpl_task_cb_num_event_kinds
+} chpl_task_cb_event_kind_t;
+
+typedef enum {
+  chpl_task_cb_info_kind_full,
+  chpl_task_cb_info_kind_id_only,
+  chpl_task_cb_num_info_kinds
+} chpl_task_cb_info_kind_t;
+
+typedef struct {
+  uint64_t nodeID;              // top-level node ID
+  chpl_task_cb_event_kind_t
+          event_kind;           // kind of event this describes
+  chpl_task_cb_info_kind_t
+          info_kind;            // kind of info in union below
+  union {
+
+    struct chpl_task_info_full {
+                                // chpl_task_cb_info_kind_full:
+      const char* filename;     //   source file of task definition
```

Comment:
Yes, I intended the 'chpl_task_cb_info_kind_*:' comments at 140 and 147 as headers, sort of.